mqtt/handler_test.go

```diff
@@ -0,0 +1,189 @@
+package mqtt
+
+import (
+	"fmt"
+	"github.com/mainflux/mainflux/pkg/uuid"
+	"github.com/stretchr/testify/require"
+	"log"
+	"testing"
+
+	"github.com/mainflux/mainflux/pkg/errors"
+
+	rdb "github.com/go-redis/redis/v8"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/mqtt/mocks"
+	"github.com/mainflux/mainflux/mqtt/redis"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/mainflux/mproxy/pkg/session"
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAuthConnect(t *testing.T) {
+	handler := newHandler()
+	cases := []struct {
+		desc    string
+		err     error
+		session *session.Client
+	}{
+		{
+			desc:    "connect without active session",
+			err:     errInvalidConnect,
+			session: nil,
+		},
+		{
+			desc: "connect when id is invalid",
+			err:  errors.New("thing identify error"),
```

Comment:
Remove this error and use one from the package. If there is no error in the package maybe you can change a bit the code and wrap the returned error with a new one.